### PR TITLE
Activitylog can use your own Activity-model now

### PR DESCRIPTION
**Pull request by Thijmen**
I wanted a feature that I was able to use my own activity model in stead of the default provided class.

Writing unit-tests are not my best point, so I tried to add some functional unit-tests. If they are inadequate, please tell me how to do this well so I can learn from it aswell :)

PS: This is my first PR ever, learnt a lot from it!

### CODE REVIEW

**Comment by freekmurze** [comment]
Hi, thanks for taking to time to put this together. It looks pretty good. 

Right now the model can be specified in `default_activity_model` in the config file. Could you rename that to `activity_model` and set the model provides by the package as the default. (of course you'll need to make changes in your code to make this work).

In that way we can avoid the special case of `activity_model` being null.

File: config/laravel-activitylog.php [hunk 1]
```diff
@@ -20,4 +20,12 @@
      * When this is null we'll use the default Laravel auth driver.
      */
     'default_auth_driver' => null,
+
+    /**
+     * You can use your own Activity-model. However, it is required to
+     * extend the Spatie\Activitylog\Models\Activity model.
+     * When this is null we'll use the default Activity model from this package.
```
Comment (freekmurze) [inline config/laravel-activitylog.php]: This comment may be removed now

File: src/ActivityLogger.php [hunk 2]
```diff
@@ -24,12 +24,17 @@ class ActivityLogger
...
+    /** @var \Spatie\Activitylog\Models\Activity */
+    protected $activityModel;
+
     public function __construct(AuthManager $auth, Repository $config)
     {
         $this->auth = $auth;
 
         $this->properties = collect();
 
+        $this->activityModel = config('laravel-activitylog.activity_model') ?? Activity::class;
```
Comment (freekmurze) [inline src/ActivityLogger.php]: throw an exception when no class or the wrong class is set

**Comment by Thijmen** [comment]
There, done 😃 .

**Comment by freekmurze** [comment]
Merge conflicts, ...

could you merge this against the current master version?

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**